src/Swarm/TUI/View.hs

```diff
@@ -865,19 +887,13 @@ drawWorld showRobots g =
 drawLoc :: Bool -> GameState -> W.Coords -> Widget Name
 drawLoc showRobots g = renderDisplay . displayLoc showRobots g
 
--- | Get the 'Display' for a specific location, by combining the
---   'Display's for the terrain, entity, and robots at the location.
-displayLoc :: Bool -> GameState -> W.Coords -> Display
-displayLoc showRobots g coords =
-  sconcat $ terrain NE.:| entity <> robots
- where
-  terrain = terrainMap M.! toEnum (W.lookupTerrain coords (g ^. world))
-  entity = maybeToList (displayForEntity <$> W.lookupEntity coords (g ^. world))
-  robots =
```

Comment:
These three lines were extracted to top-level functions so they can be re-used in my new cursor rendering logic.